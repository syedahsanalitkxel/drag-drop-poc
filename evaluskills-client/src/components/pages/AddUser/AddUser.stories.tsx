import React from 'react';

import { storiesOf } from '@storybook/react';

import Index from '.';

storiesOf('Index', module).add('Index', () => <Index />);
