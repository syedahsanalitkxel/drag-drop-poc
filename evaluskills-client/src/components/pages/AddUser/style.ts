export const styles = {
  btn: {
    marginLeft: 2,
    marginRight: 2,
  },
  modal_width: {
    maxWidth: '870px',
  },
};
