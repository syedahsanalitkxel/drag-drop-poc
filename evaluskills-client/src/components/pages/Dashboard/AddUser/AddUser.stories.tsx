import React from 'react';

import { storiesOf } from '@storybook/react';

import AddUser from './AddUser';

storiesOf('AddUser', module).add('AddUser', () => <AddUser />);
