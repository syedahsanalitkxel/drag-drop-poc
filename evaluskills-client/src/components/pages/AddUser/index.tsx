import { Field, Formik } from 'formik';
import React, { useContext, useState } from 'react';
import { Button, FormFeedback, Input, Modal, ModalFooter, ModalHeader } from 'reactstrap';
import * as Yup from 'yup';
import { styles } from './style';
import PageBody from '../../atoms/PageBody';
import styled from 'styled-components';
import FormikBag from '../../../interfaces/FormikBag';
import { lookups } from '../../../modules/Lookup/enum';
import LookupContext, { LookupContextConsumer } from '../../../modules/Lookup/context';
import FormElement, { FormElementTypes } from '../../molecules/FormElement';

interface ModalProps {
  visible?: boolean;
  toggle: () => void;
  submitHandler: (values: any) => void;
  name?: string;
  FormValues: any;
  cancelHandler: () => void;
}

const StyledButton = styled(Button)`
  margin-left: 20px;
  margin-right: 5px;
`;

export const AddUser: React.FunctionComponent<ModalProps> = ({
  visible,
  toggle,
  name,
  FormValues,
  submitHandler,
  cancelHandler,
}) => {
  const { findKey } = useContext(LookupContext);
  const [disabled, setDisabled] = useState(false);

  function submitForm(values: any) {
    submitHandler(values);
  }

  const addUserSchema = Yup.object().shape({
    email: Yup.string()
      .required()
      .email(),
    firstName: Yup.string()
      .min(2, 'Too Short!')
      .max(50, 'Too Long!')
      .matches(/^[a-zA-Z ]+$/, 'Name Must be in Alphabets')
      .required('Required Field'),
    lastName: Yup.string()
      .min(2, 'Too Short!')
      .max(50, 'Too Long!')
      .matches(/^[a-zA-Z ]+$/, 'Name Must be in Alphabets')
      .required('Required Field'),
    role: Yup.string()
      .min(2, 'Too Short!')
      .max(50, 'Too Long!')
      .required('Required'),
  });

  function renderUserRoleDropdown() {
    if (findKey) {
      return findKey(lookups.userRolesLookUp).map(application => {
        return (
          <option value={application.text} key={application.value}>
            {application.text}
          </option>
        );
      });
    }
  }

  const renderForm = (formikprops: FormikBag) => (
    <form onSubmit={formikprops.handleSubmit} className={'form'}>
      <div className=" ">
        <div className="no-borders">
          <div className="form-group row">
            <div className="col-sm-6">
              <label className="col-sm-10 col-form-label font-bold">First Name</label>
              <div className="col-sm-10">
                <Input
                  type="text"
                  name="firstName"
                  className="form-control"
                  placeholder="First Name"
                  tag={Field}
                  invalid={!!(formikprops.touched.firstName && formikprops.errors.firstName)}
                />

                <FormFeedback tooltip={true}>{formikprops.errors.firstName}</FormFeedback>
              </div>
            </div>
            <div className="col-sm-6">
              <label className="col-sm-10 col-form-label font-bold">Last Name</label>
              <div className="col-sm-10">
                <Input
                  type="text"
                  name="lastName"
                  className="form-control"
                  placeholder="Last Name"
                  tag={Field}
                  invalid={!!(formikprops.touched.lastName && formikprops.errors.lastName)}
                />

                <FormFeedback tooltip={true}>{formikprops.errors.lastName}</FormFeedback>
              </div>
            </div>
          </div>
          <div className="hr-line-dashed" />
          <div className="form-group row">
            <div className="col-sm-6">
              <label className="col-sm-10 col-form-label font-bold">Email</label>
              <div className="col-sm-10">
                {name === 'Edit' ? (
                  <Input
                    type="text"
                    name="email"
                    className="form-control"
                    disabled={true}
                    placeholder="Email"
                    tag={Field}
                    invalid={!!(formikprops.touched.email && formikprops.errors.email)}
                  />
                ) : (
                  <Input
                    type="text"
                    name="email"
                    className="form-control"
                    placeholder="Email"
                    tag={Field}
                    invalid={!!(formikprops.touched.email && formikprops.errors.email)}
                  />
                )}

                <FormFeedback tooltip={true}>{formikprops.errors.email}</FormFeedback>
              </div>
            </div>
            <div className="col-sm-6">
              <div className="col-sm-10">
                <FormElement
                  label="Role"
                  name="role"
                  formikprops={formikprops}
                  type={FormElementTypes.SELECT}
                  inline={true}
                  last={true}
                >
                  <option value=""> Select One</option>
                  {renderUserRoleDropdown()}
                </FormElement>
              </div>
            </div>
          </div>
        </div>
      </div>

      <PageBody>
        <div className="row m-b-25">
          <StyledButton type="button" size="lg" onClick={cancelHandler}>
            Cancel
          </StyledButton>
          <StyledButton type="submit" color="primary" size="lg">
            Save
          </StyledButton>
        </div>
      </PageBody>
    </form>
  );

  return (
    <Modal isOpen={visible} toggle={toggle} style={styles.modal_width}>
      <ModalHeader toggle={toggle}>{name} User</ModalHeader>
      <Formik
        enableReinitialize={true}
        initialValues={FormValues}
        validationSchema={addUserSchema}
        onSubmit={submitForm}
      >
        {formikprops => renderForm(formikprops)}
      </Formik>
    </Modal>
  );
};

AddUser.defaultProps = {
  visible: false,
};

export default AddUser;
