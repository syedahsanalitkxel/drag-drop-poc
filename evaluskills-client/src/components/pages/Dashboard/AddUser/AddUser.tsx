import { Field, Formik } from 'formik';
import React from 'react';
import { FormFeedback, Input } from 'reactstrap';
import * as Yup from 'yup';
import DashboardTemplate from '../../../templates/DashboardTemplate';
import { styles } from '../style';

interface MyFormValues {
  firstName: string;
  lastName: string;
  email: string;
  role: string;
}

export const AddUser: React.FunctionComponent = () => {
  const addUserSchema = Yup.object().shape({
    email: Yup.string().email(),
    firstName: Yup.string()
      .min(2, 'Too Short!')
      .max(250, 'Too Long!')
      .required('Required'),
    lastName: Yup.string()
      .min(2, 'Too Short!')
      .max(50, 'Too Long!')
      .required('Required'),
    role: Yup.string()
      .min(2, 'Too Short!')
      .max(50, 'Too Long!')
      .required('Required'),
  });

  const renderForm = (formikprops: any) => (
    <form onSubmit={formikprops.handleSubmit} className={'form'}>
      <div className="wrapper wrapper-content animated fadeInRight">
        <div className="ibox ">
          <div className="ibox-content no-borders">
            <div className="form-group row">
              <div className="col-sm-4">
                <label className="col-sm-10 col-form-label font-bold">First Name</label>
                <div className="col-sm-10">
                  <Input
                    type="text"
                    name="firstName"
                    className="form-control"
                    tag={Field}
                    invalid={!!(formikprops.touched.firstName && formikprops.errors.firstName)}
                  />

                  <FormFeedback tooltip={true}>{formikprops.errors.firstName}</FormFeedback>
                </div>
              </div>
              <div className="col-sm-4">
                <label className="col-sm-10 col-form-label font-bold">Last Name</label>
                <div className="col-sm-10">
                  <Input
                    type="text"
                    name="lastName"
                    className="form-control"
                    tag={Field}
                    invalid={!!(formikprops.touched.lastName && formikprops.errors.lastName)}
                  />

                  <FormFeedback tooltip={true}>{formikprops.errors.lastName}</FormFeedback>
                </div>
              </div>
            </div>
            <div className="hr-line-dashed" />
            <div className="form-group row">
              <div className="col-sm-4">
                <label className="col-sm-10 col-form-label font-bold">Email</label>
                <div className="col-sm-10">
                  <Input
                    type="text"
                    name="email"
                    className="form-control"
                    tag={Field}
                    invalid={!!(formikprops.touched.email && formikprops.errors.email)}
                  />

                  <FormFeedback tooltip={true}>{formikprops.errors.email}</FormFeedback>
                </div>
              </div>
              <div className="col-sm-4">
                <label className="col-sm-10 col-form-label font-bold">Role</label>
                <div className="col-sm-10">
                  <Input
                    type="text"
                    name="role"
                    className="form-control"
                    tag={Field}
                    invalid={!!(formikprops.touched.role && formikprops.errors.role)}
                  />

                  <FormFeedback tooltip={true}>{formikprops.errors.role}</FormFeedback>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="m-t-15 m-b-15">
          <button type="button" style={styles.btn} className="btn btn-default btn-lg">
            Cancel
          </button>
          <button
            type="button"
            style={styles.btn}
            id={'submit'}
            name="submit"
            className="btn btn-primary btn-lg"
          >
            Save
          </button>
          <button type="button" style={styles.btn} className="btn btn-primary btn-lg">
            Save &amp; Add More
          </button>
        </div>
      </div>
    </form>
  );

  return (
    <DashboardTemplate>
      <Formik
        initialValues={{ firstName: '', lastName: '', email: '', role: '' }}
        validationSchema={addUserSchema}
        onSubmit={() => {}}
      >
        {formikprops => renderForm(formikprops)}
      </Formik>
    </DashboardTemplate>
  );
};

export default AddUser;
