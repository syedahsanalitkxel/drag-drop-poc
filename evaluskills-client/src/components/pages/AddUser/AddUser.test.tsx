import { configure, mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import AddUser from './index';

configure({ adapter: new Adapter() });

interface ModalProps {
  visible?: boolean;
  toggle: () => void;
  name?: string;
  FormValues: any;
}

describe('MyComponent', () => {
  const toggleModal = jest.fn();
  const modalValues = {
    email: 'maria@evaluskills.com',
    firstName: 'maria',
    id: '1',
    lastName: 'gracia',
    role: 'admin',
  };
  const props: ModalProps = {
    visible: true,
    toggle: toggleModal,
    name: 'Add',
    FormValues: modalValues,
  };

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AddUser {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <AddUser {...props} />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('to be defined of form control `.form`s', () => {
    const wrapper = shallow(<AddUser {...props} />);
    expect(wrapper.find('.form')).toBeDefined();
  });
  it('4 Input fields in `.form`', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.find('Input')).toHaveLength(4);
  });
  it('4 Form Feedback in `.form`', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.find('FormFeedback')).toHaveLength(4);
  });
  it('16 form control `.form-control`s', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.find('.form-control')).toHaveLength(16);
    expect(wrapper.find('.form-control')).toBeDefined();
  });
  it('Find Modal is running successfully ', () => {
    const wrapper = shallow(<AddUser {...props} />);
    expect(wrapper.find('Modal')).toBeDefined();
  });
  it('First Name label exist ', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.text()).toContain('First Name');
  });
  it('Last Name label exist ', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.text()).toContain('Last Name');
  });
  it('Email label exist ', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.text()).toContain('Email');
  });
  it('Role label exist ', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    expect(wrapper.text()).toContain('Role');
  });

  it('submit form ', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    wrapper.find('form').simulate('submit');
  });
  it('submit form ', () => {
    const wrapper = mount(
      <Router>
        <AddUser {...props} />
      </Router>
    );
    wrapper.find('form').simulate('submit', {
      preventDefault: (props: any) => {
        console.log('propssss', props);
      },
    });
  });

  describe('Component functionality', () => {
    it('should not submit if required fields are empty', () => {
      const badFormValues = {
        email: 'aa',
        firstName: ' ',
        id: ' ',
        lastName: ' ',
        role: ' ',
      };
      const badprops: ModalProps = {
        visible: true,
        toggle: toggleModal,
        name: 'Add',
        FormValues: badFormValues,
      };
      const resetForm = jest.fn();
      const componentRender = mount(
        <Router>
          <AddUser {...badprops} />
        </Router>
      );
      componentRender
        .find('Formik')
        .instance()
        .props.onSubmit(badFormValues, { resetForm });
      // console.log(componentRender.find('FormFeedback').debug());
      // console.log(componentRender.update().find('.validation-error'));
      // console.log(componentRender.find('Formik').instance());
      // expect(componentRender.find('.validation-error').text()).toEqual('Company Name is required');
    });
  });
});
