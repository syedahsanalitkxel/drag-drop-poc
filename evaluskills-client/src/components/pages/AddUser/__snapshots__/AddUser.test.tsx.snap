// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<DashboardTemplate>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "email": "",
        "firstName": "",
        "lastName": "",
        "role": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "role",
          "lastName",
          "firstName",
          "email",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "lastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "role": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Component />
  </Formik>
</DashboardTemplate>
`;
